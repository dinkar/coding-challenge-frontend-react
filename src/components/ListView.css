.incidents__list {
  list-style: none;
  margin-bottom: 0;
  padding: 0;
  max-width: 100%;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: stretch; 
}

.incidents__item {
  margin-bottom: 1.5rem;
  padding: 1.5rem;
  border: solid 2px #6b7581;
  background: #dadcdf;
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
}

.incidents__item img {
  width: 20%;
}

.incidents__item--description {
  width: 80%;
  margin-left: 1.5rem;
}

.incidents__item div{
  margin-bottom: 1rem;
}

