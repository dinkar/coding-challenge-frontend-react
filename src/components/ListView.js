import React from 'react';
import { Link } from 'react-router-dom';
import './ListView.css';

const listView = props => (
  <ul className="incidents__list">
    {props.items.map(item => (
      <li key={item.id} className="incidents__item">
        <img
          src={
            item.media.image_url_thumb
              ? item.media.image_url_thumb
              : process.env.PUBLIC_URL + '/bike-image-missing.jpg'
          }
          alt={item.title}
          onError={e => {
            e.target.onerror = null;
            e.target.src = process.env.PUBLIC_URL + '/bike-image-missing.jpg';
          }}
        />
        <div className="incidents__item--description">
          <Link to={`/case/${item.id}`} style={{ marginBottom: '1rem' }}>
            <h3>{item.title}</h3>
          </Link>
          <div>
            {item.description ? item.description : 'DESCRIPTION MISSING'}
          </div>
          <div>
            {new Date(item.occurred_at * 1000).toLocaleString()} -{' '}
            {item.address}
          </div>
        </div>
      </li>
    ))}
  </ul>
);

export default listView;
