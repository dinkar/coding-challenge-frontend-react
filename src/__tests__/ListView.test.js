import ListView from '../components/ListView';
import renderer from 'react-test-renderer';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';

const items1 = [
  {
    id: 72790,
    title: 'Stolen 2014 Flyer Wave(silver or gray)',
    description:
      'having seen a man interested in this e-bike, but hurrying away, when I came out of the house. 20 later the bike has gone.',
    address: 'Berlin, 10779, DE',
    occurred_at: 1495692000,
    updated_at: 1550446746,
    media: {
      image_url_thumb: null
    }
  },
  {
    id: 4853,
    title: 'Stolen 2005 Trek 4300 Disc (silver or gray)',
    description:
      'Bike was parked in a locked cellar compartment. During night the door of the compartment was forced open and two bikes were stolen: Trek 4300 Disc and Trek Fuel Ex 8.',
    address: 'Halle, 06120, DE',
    occurred_at: 1410760800,
    updated_at: 1550387537,
    media: {
      image_url_thumb:
        'https://bikebook.s3.amazonaws.com/uploads/Fr/112/small_Asset_140338'
    }
  }
];

const items2 = [
  {
    id: 4853,
    title: 'Stolen 2005 Trek 4300 Disc (silver or gray)',
    description:
      'Bike was parked in a locked cellar compartment. During night the door of the compartment was forced open and two bikes were stolen: Trek 4300 Disc and Trek Fuel Ex 8.',
    address: 'Halle, 06120, DE',
    occurred_at: 1410760800,
    updated_at: 1550387537,
    media: {
      image_url:
        'https://bikebook.s3.amazonaws.com/uploads/Fr/112/Asset_140338',
      image_url_thumb:
        'https://bikebook.s3.amazonaws.com/uploads/Fr/112/small_Asset_140338'
    }
  },
  {
    id: 4879,
    title: 'Stolen 2014 Trek Fuel EX 8 26 (black)',
    description:
      'Bike was parked in a locked cellar compartment. During night the door of the comparment was forced open and two bikes wer stolen: Trek Fuel EX 8 and Trek 4300 Disc. ',
    address: 'Halle, 06120, DE',
    occurred_at: 1410760800,
    updated_at: 1550387699,
    media: {
      image_url:
        'https://bikebook.s3.amazonaws.com/uploads/Fr/112/Asset_140338',
      image_url_thumb:
        'https://bikebook.s3.amazonaws.com/uploads/Fr/112/small_Asset_140338'
    }
  },
  {
    id: 3522,
    title: 'Stolen 2012 Cannondale Caad8 7 Sora (black and white)',
    description:
      'Stolen during the day from yard bike stands between three offices in front of restaurant.',
    address: 'Berlin, 10115, DE',
    occurred_at: 1408341600,
    updated_at: 1550387281,
    media: {
      image_url:
        'https://bikebook.s3.amazonaws.com/uploads/Fr/9764/c_13_3RA87C_gry_14.png',
      image_url_thumb:
        'https://bikebook.s3.amazonaws.com/uploads/Fr/9764/small_c_13_3RA87C_gry_14.png'
    }
  }
];

describe('<ListView> tests', () => {
  it('should match with the snapshot', () => {
    const component = renderer.create(
      <MemoryRouter>
        <ListView items={items1} />
      </MemoryRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.items = items2;
    expect(tree).toMatchSnapshot();
  });
});
