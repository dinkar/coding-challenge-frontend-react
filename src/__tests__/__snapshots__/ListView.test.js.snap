// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListView> tests should match with the snapshot 1`] = `
<ul
  className="incidents__list"
>
  <li
    className="incidents__item"
  >
    <img
      alt="Stolen 2014 Flyer Wave(silver or gray)"
      onError={[Function]}
      src="/bike-image-missing.jpg"
    />
    <div
      className="incidents__item--description"
    >
      <a
        href="/case/72790"
        onClick={[Function]}
        style={
          Object {
            "marginBottom": "1rem",
          }
        }
      >
        <h3>
          Stolen 2014 Flyer Wave(silver or gray)
        </h3>
      </a>
      <div>
        having seen a man interested in this e-bike, but hurrying away, when I came out of the house. 20 later the bike has gone.
      </div>
      <div>
        Thu May 25 2017
         - 
        Berlin, 10779, DE
      </div>
    </div>
  </li>
  <li
    className="incidents__item"
  >
    <img
      alt="Stolen 2005 Trek 4300 Disc (silver or gray)"
      onError={[Function]}
      src="https://bikebook.s3.amazonaws.com/uploads/Fr/112/small_Asset_140338"
    />
    <div
      className="incidents__item--description"
    >
      <a
        href="/case/4853"
        onClick={[Function]}
        style={
          Object {
            "marginBottom": "1rem",
          }
        }
      >
        <h3>
          Stolen 2005 Trek 4300 Disc (silver or gray)
        </h3>
      </a>
      <div>
        Bike was parked in a locked cellar compartment. During night the door of the compartment was forced open and two bikes were stolen: Trek 4300 Disc and Trek Fuel Ex 8.
      </div>
      <div>
        Mon Sep 15 2014
         - 
        Halle, 06120, DE
      </div>
    </div>
  </li>
</ul>
`;

exports[`<ListView> tests should match with the snapshot 2`] = `
<ul
  className="incidents__list"
  items={
    Array [
      Object {
        "address": "Halle, 06120, DE",
        "description": "Bike was parked in a locked cellar compartment. During night the door of the compartment was forced open and two bikes were stolen: Trek 4300 Disc and Trek Fuel Ex 8.",
        "id": 4853,
        "media": Object {
          "image_url": "https://bikebook.s3.amazonaws.com/uploads/Fr/112/Asset_140338",
          "image_url_thumb": "https://bikebook.s3.amazonaws.com/uploads/Fr/112/small_Asset_140338",
        },
        "occurred_at": 1410760800,
        "title": "Stolen 2005 Trek 4300 Disc (silver or gray)",
        "updated_at": 1550387537,
      },
      Object {
        "address": "Halle, 06120, DE",
        "description": "Bike was parked in a locked cellar compartment. During night the door of the comparment was forced open and two bikes wer stolen: Trek Fuel EX 8 and Trek 4300 Disc. ",
        "id": 4879,
        "media": Object {
          "image_url": "https://bikebook.s3.amazonaws.com/uploads/Fr/112/Asset_140338",
          "image_url_thumb": "https://bikebook.s3.amazonaws.com/uploads/Fr/112/small_Asset_140338",
        },
        "occurred_at": 1410760800,
        "title": "Stolen 2014 Trek Fuel EX 8 26 (black)",
        "updated_at": 1550387699,
      },
      Object {
        "address": "Berlin, 10115, DE",
        "description": "Stolen during the day from yard bike stands between three offices in front of restaurant.",
        "id": 3522,
        "media": Object {
          "image_url": "https://bikebook.s3.amazonaws.com/uploads/Fr/9764/c_13_3RA87C_gry_14.png",
          "image_url_thumb": "https://bikebook.s3.amazonaws.com/uploads/Fr/9764/small_c_13_3RA87C_gry_14.png",
        },
        "occurred_at": 1408341600,
        "title": "Stolen 2012 Cannondale Caad8 7 Sora (black and white)",
        "updated_at": 1550387281,
      },
    ]
  }
>
  <li
    className="incidents__item"
  >
    <img
      alt="Stolen 2014 Flyer Wave(silver or gray)"
      onError={[Function]}
      src="/bike-image-missing.jpg"
    />
    <div
      className="incidents__item--description"
    >
      <a
        href="/case/72790"
        onClick={[Function]}
        style={
          Object {
            "marginBottom": "1rem",
          }
        }
      >
        <h3>
          Stolen 2014 Flyer Wave(silver or gray)
        </h3>
      </a>
      <div>
        having seen a man interested in this e-bike, but hurrying away, when I came out of the house. 20 later the bike has gone.
      </div>
      <div>
        Thu May 25 2017
         - 
        Berlin, 10779, DE
      </div>
    </div>
  </li>
  <li
    className="incidents__item"
  >
    <img
      alt="Stolen 2005 Trek 4300 Disc (silver or gray)"
      onError={[Function]}
      src="https://bikebook.s3.amazonaws.com/uploads/Fr/112/small_Asset_140338"
    />
    <div
      className="incidents__item--description"
    >
      <a
        href="/case/4853"
        onClick={[Function]}
        style={
          Object {
            "marginBottom": "1rem",
          }
        }
      >
        <h3>
          Stolen 2005 Trek 4300 Disc (silver or gray)
        </h3>
      </a>
      <div>
        Bike was parked in a locked cellar compartment. During night the door of the compartment was forced open and two bikes were stolen: Trek 4300 Disc and Trek Fuel Ex 8.
      </div>
      <div>
        Mon Sep 15 2014
         - 
        Halle, 06120, DE
      </div>
    </div>
  </li>
</ul>
`;
